components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -218,6 +225,50 @@ func (wsc *WorkspaceController) handleWorkspaceInit(ctx context.Context, ws *wor
 	return ctrl.Result{}, nil
 }
 
+func initializerMetricsFromInitializerStats(stats *csapi.InitializerMetrics) *workspacev1.InitializerMetrics {
+	if stats == nil {
+		return nil
+	}
+
+	result := workspacev1.InitializerMetrics{}
+	for _, metric := range *stats {
+		switch metric.Type {
+		case "git":
+			result.Git = &workspacev1.InitializerStepMetric{
+				Duration: &metav1.Duration{Duration: metric.Duration},
+				Size:     metric.Size,
+			}
+		case "fileDownload":
+			result.FileDownload = &workspacev1.InitializerStepMetric{
+				Duration: &metav1.Duration{Duration: metric.Duration},
+				Size:     metric.Size,
+			}
+		case "snapshot":
+			result.Snapshot = &workspacev1.InitializerStepMetric{
+				Duration: &metav1.Duration{Duration: metric.Duration},
+				Size:     metric.Size,
+			}
+		case "fromBackup":
```

Comment:
The type here seems to check out with `components/content-service-api/initializer.proto`, but I couldn't get it to pass any metrics down when I do a start of a stopped workspace.